### PR TITLE
Make AutoCrusher aware of Cloak and Disguise

### LINKED ISSUE #21102: AutoCrusher ignore "RequiresCondition"
Version: playtest-20230927

test case:
[test.zip](https://github.com/OpenRA/OpenRA/files/12841760/test.zip)

### PR DESCRIPTION
**Pull request by abcdefg30**
Reverts #21104 as that was a bit premature: https://github.com/OpenRA/OpenRA/pull/21104#issuecomment-1755269773

In exchange resolves a different todo and fixes #21102.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/AutoCrusher.cs [hunk 1]
```diff
@@ -50,20 +51,17 @@ public AutoCrusher(Actor self, AutoCrusherInfo info)
 			moveInfo = self.Info.TraitInfo<IMoveInfo>();
 			nextScanTime = self.World.SharedRandom.Next(Info.MinimumScanTimeInterval, Info.MaximumScanTimeInterval);
 			isAircraft = move is Aircraft;
+			ignoresDisguise = self.Info.HasTraitInfo<IgnoresDisguiseInfo>();
 		}
 
 		void INotifyIdle.TickIdle(Actor self)
 		{
-			if (nextScanTime-- > 0)
+			if (IsTraitDisabled || IsTraitPaused || nextScanTime-- > 0)
```
Comment (PunkPun) [inline OpenRA.Mods.Common/Traits/AutoCrusher.cs]: Usually pause does different behaviour. Here there's no difference. Perhaps downgrade to conditional trait?

Comment (abcdefg30) [reply OpenRA.Mods.Common/Traits/AutoCrusher.cs]: Hm, good point. Fixed. 👍

File: OpenRA.Mods.Common/Traits/AutoCrusher.cs [hunk 2]
```diff
@@ -50,20 +51,17 @@ public AutoCrusher(Actor self, AutoCrusherInfo info)
 			moveInfo = self.Info.TraitInfo<IMoveInfo>();
 			nextScanTime = self.World.SharedRandom.Next(Info.MinimumScanTimeInterval, Info.MaximumScanTimeInterval);
 			isAircraft = move is Aircraft;
+			ignoresDisguise = self.Info.HasTraitInfo<IgnoresDisguiseInfo>();
 		}
 
 		void INotifyIdle.TickIdle(Actor self)
 		{
-			if (nextScanTime-- > 0)
+			if (IsTraitDisabled || nextScanTime-- > 0)
```
Comment (PunkPun) [inline OpenRA.Mods.Common/Traits/AutoCrusher.cs:58]: Ah and one last thing, on being enabled to should set nextScanTime to 0

Comment (abcdefg30) [reply OpenRA.Mods.Common/Traits/AutoCrusher.cs:58]: Good spot, fixed.

**Review by PunkPun — APPROVED** [review]

**Comment by PunkPun** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/c88a7456ecd12a85e6f5d92e2ca46353bbb67bc3...36d78107bb0e063a1dce216ddc17f549ae207de9)

**Pull request merged**